src/pandas_profiling/__init__.py

```diff
@@ -271,3 +330,146 @@ def to_app(self):
         app_widgets = QtReport(self.report).render()
 
         app = get_app(app, self.title, app_widgets)
+
+    def _repr_html_(self):
+        """The ipython notebook widgets user interface gets called by the jupyter notebook."""
+        self.to_notebook_iframe()
+
+    def __repr__(self):
+        """Override so that Jupyter Notebook does not print the object."""
+        return ""
+
+    def dumps(self) -> bytes:
+        """
+        Serialize ProfileReport and return bytes for reproducing ProfileReport or Caching.
+
+        Returns:
+            Bytes which contains hash of DataFrame, config, sample, _description_set and _report
+        """
+        import pickle
```

Comment:
Why choose pickle? Have you considered faster alternatives?

https://www.benfrederickson.com/dont-pickle-your-data/

Reply:
I have considered about other methods of serialization and deserialization, the central paradox is that report and description_set contains many types of object.
 - Built-in types: `bool`, `int`, `str`, `float`, `None`: Can be handled by json
 - Custom types: `Message`,`Renderable`,`Sequence`: Can be handled by mixin class you have mentioned.
 - Enum: `MessageType`,`Variable`: Need some function and wrapper to handle it
 - Custom types in dependent package : `np.int64`,`np.float64`,`np.bool_`,`np.ndarray`,`pd.Series`,`datetime`: Need wrapper class to handle correspondingly.

Instead of pickle, I think the alternative is to design a special data structure instead of a dictionary to store and handle the data. I chose pickle because the usage of report and  description_set are so broad that changing the core data structure might cause instability.

Reply:
For now pickle looks like the most stable option indeed.